File: app/src/main/java/me/pwcong/jpstart/mvp/model/JPStartTabFragmentModelImpl.java
package me.pwcong.jpstart.mvp.model;

import java.util.ArrayList;
import java.util.List;

import me.pwcong.jpstart.App;
import me.pwcong.jpstart.R;
import me.pwcong.jpstart.conf.Constants;
import me.pwcong.jpstart.mvp.bean.JPTab;
import me.pwcong.jpstart.utils.ResourceUtils;

/**
 * Created by Pwcong on 2016/9/27.
 */

public class JPStartTabFragmentModelImpl implements BaseModel.JPStartTabFragmentModel {

    @Override
    public List<JPTab> getData() {

        List<JPTab> list=new ArrayList<>();
        String hiragana = ResourceUtils.getString(App.getInstance(), R.string.hiragana);
        String katakana = ResourceUtils.getString(App.getInstance(), R.string.katakana);

        list.add(new JPTab(Constants.TYPE_HIRAGANA,hiragana));
        list.add(new JPTab(Constants.TYPE_KATAKANA,katakana));

        return list;
    }
}

File: app/src/main/java/com/github/pwcong/jpstart/mvp/model/impl/JPStartTabFragmentModelImpl.kt
package com.github.pwcong.jpstart.mvp.model.impl

import com.github.pwcong.jpstart.App
import com.github.pwcong.jpstart.R
import com.github.pwcong.jpstart.constants.Constants
import com.github.pwcong.jpstart.mvp.bean.JPTab
import com.github.pwcong.jpstart.mvp.model.BaseModel.JPStartTabFragmentModel
import com.github.pwcong.jpstart.utils.ResourceUtils

class JPStartTabFragmentModelImpl : JPStartTabFragmentModel {
    override val data: List<JPTab>
        get() {
            val list: MutableList<JPTab> = ArrayList()
            val qingyin: String = ResourceUtils.getString(App.getInstance(), R.string.qingyin)
            val zhuoyin: String = ResourceUtils.getString(App.getInstance(), R.string.zhuoyin)
            val aoyin: String = ResourceUtils.getString(App.getInstance(), R.string.aoyin)

            list.add(JPTab(Constants.CATEGORY_QINGYIN, qingyin))
            list.add(JPTab(Constants.CATEGORY_ZHUOYIN, zhuoyin))
            list.add(JPTab(Constants.CATEGORY_AOYIN, aoyin))

            return list
        }
}
